src/server/helpers/entityRouteUtils.js

```diff
@@ -53,3 +64,33 @@ export function generateEntityProps(
 
 	return generateProps(req, res, props);
 }
+
+/**
+ * Return markup for the entity editor.
+ * @param {object} props - react props
+ * @param {function} rootReducer - redux root reducer
+ * @returns {string} - HTML string
+ */
+export function entityEditorMarkup(
+	props: { initialState: Object,
+			 entityType: string },
+	rootReducer: Function) {
+	const {initialState, ...rest} = props;
+	const store = createStore(rootReducer, Immutable.fromJS(initialState));
+	const EntitySection = getEntitySection(props.entityType);
+	const markup = ReactDOMServer.renderToString(
+		<Layout {...propHelpers.extractLayoutProps(rest)}>
+			<Provider store={store}>
+				<EntityEditor
+					validate={getValidator(props.entityType)}
+					{...propHelpers.extractChildProps(rest)}
+				>
+					<EntitySection/>
+				</EntityEditor>
+			</Provider>
+		</Layout>
+	);
+	props.initialState = store.getState();
```

Comment:
It might be a good idea to document that props gets modified within this function. Might be better to return updated props as well as markup to make this function cleaner.

Reply:
I'll document it, but I don't agree that returning `props` would make it cleaner. The value is unrelated to markup, and it'll make calling code messier without any counter-benefit.

Reply:
On second thought, this would allow me to declare `props` as the result of `entityEditorMarkup`. Will do this.